cf_pandas/utils.py

```diff
@@ -30,27 +30,31 @@ def astype(value, type_):
     return value
 
 
-def set_up_criteria(criteria: Optional[dict] = None):
+def set_up_criteria(criteria: Union[dict, Iterable] = None) -> ChainMap:
     """Get custom criteria from options.
 
     Parameters
     ----------
     criteria : dict, optional
-        Criteria to use to map from variable to attributes describing the variable. If user has defined custom_criteria, this will be used by default.
+        Criteria to use to map from variable to attributes describing the variable. If user has defined
+        custom_criteria, this will be used by default.
+
+    Returns
+    -------
+    ChainMap
+        Criteria
     """
 
     if criteria is None:
         if not OPTIONS["custom_criteria"]:
-            return []
-        criteria = OPTIONS["custom_criteria"]
-
-    if criteria is not None:
-        criteria = always_iterable(criteria, allowed=(tuple, list, set))
-
-    criteria = always_iterable(criteria, allowed=(tuple, list, set))
-    criteria = ChainMap(*criteria)
-
-    return criteria
+            raise KeyError(
```

Comment:
This should probably be a `ValueError` since the exception is due to an unsupported value being passed to the method, in lieu of a key not existing in a dict.